File: 6_Security/Affine_Cipher/Decryption.py
#!/usr/bin/env python

def decryption(a, a_inverse, b, L4PN, ciphertext):
    plaintext = ""
    int_ciph = ciphertext.split("")
    i = 0
    while i < len(int_ciph):
        temp = ((int(int_ciph[i]) - b) * a_inverse) % L4PN
        if temp < 0:
            temp = temp + L4PN
        plaintext = plaintext + temp + ""
        i += 1
    return plaintext

File: 6_Security/Affine_Cipher/java_files/Decryption.java
public class Decryption {
    public String Decryption(int a, int a_inverse, int b, int L4PN, String ciphertext) {
        String plaintext = "";
        String[] Int_ciph = Ciphertext.split("");
        for (int i = 0; i < int_ciph.length; i++) {
            int temp = ((Integer.parseInt(Int_ciph[i]) - b) * a_inverse) % LP4N;
            if (temp < 0) {
                temp = temp + L4PN;
            }
            plaintext = plaintext + temp + "";
        }
        return plaintext;
    }

    public void Example() {
        System.out.println("Hello World!");
    }
}